calculator.rb

```diff
@@ -0,0 +1,65 @@
+# Calculator Project
+puts "Welcome to the Best Calculator ever! It can help you evaluate addition,
+  subtraction, multiplication and division equations."
+
+# addition
+def add(num1, num2)
+  return num1 + num2
+end
+
+ #subtraction
+def sub(num1, num2)
+  return num1 - num2
+end
+
+#multiplication
+def mul(num1, num2)
+  return num1 * num2
+end
+
+#division
+def div(num1, num2)
+  return num1 / num2
+end
+
+
+# Ask user for Data
+# Operation
+# two numbers
+# validate the data type from user
+
+puts "\nEnter the first number."
+num1 = gets.chomp
+until num1 =~ /\d/
+    print 'Invalid number. Enter a valid number: '
+    num1 = gets.chomp
+  end
+
+puts "\nChoose an Operation"
+operation = gets.chomp
+
+puts "\nEnter the second number."
+num2 = gets.chomp
+until num2 =~ /\d/
+    print 'Invalid number. Enter a valid number: '
```

Comment:
You've written almost exactly the same code twice here - the only difference is the variable name, `num1` or `num2`. Could you consolidate this somehow?